File: week1/CamelCase4.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

/*
S;M;plasticCup()

C;V;mobile phone

C;C;coffee machine

S;C;LargeSoftwareBook

C;M;white sheet of paper

S;V;pictureFrame
*/

class Conversor
{
private:
    char operation;
    char method;
    string text;

public:
    char getOperation(void)
    {
        return operation;
    }

    void setOperation(char operation)
    {
        this->operation = operation;
    }

    char getMethod(void)
    {
        return method;
    }

    void setMethod(char method)
    {
        this->method = method;
    }

    string getText(void)
    {
        return text;
    }

    void setText(string text)
    {
        this->text = text;
    }

    string convert(char operation) {
        string newText;

        if (operation == 'S')
            cout << "S" << endl;
            // newText = strip(method, text);
        else if (operation == 'C')
            newText = combine(this->method, this->text);

        return newText;
    }

    string strip(char method, string text) {
        if (method == 'V') {

        } else if (method == 'M') {

        } else if (method == 'C') {

        }

        return NULL;
    }

    string combine(char method, string text) {
        string tmp;

        if (method == 'C')
            text[0] = toupper(text[0]);
        else if(method == 'M')
            text.append("()");

        tmp.push_back(text[0]);
        
        for (int i = 1; i < text.length(); i++)
        {
            if (text[i - 1] == ' ') {
                tmp.push_back(toupper(text[i]));
            } else if (text[i] != ' ') {
                tmp.push_back(text[i]);
            }            
        }
        
        return tmp;
    }
};

int main()
{
    string input;

    getline(cin, input);
    
    Conversor conversor;

    conversor.setOperation(input[0]);
    conversor.setMethod(input[2]);
    conversor.setText(input.substr(4, input.length() - 1));

    cout << conversor.convert(conversor.getOperation());

    return 0;
}

File: week1/CamelCase4.java
import java.util.Scanner;

class CamelCase4 {
    public char method;
    public char operation;
    public String text;

    public char getMethod() {
        return method;
    }

    public void setMethod(char method) {
        this.method = method;
    }

    public char getOperation() {
        return operation;
    }

    public void setOperation(char operation) {
        this.operation = operation;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public String strip() {
        String tmp = "";

        if (method == 'C')
            text = text.substring(0, 1).toLowerCase() + text.substring(1);
        else if (method == 'M') {
            text = text.substring(0, 1).toLowerCase() + text.substring(1, text.length() - 2);
        }

        for (int i = 0; i < text.length(); i++) {
            char letter = text.charAt(i);
            if (Character.isUpperCase(letter)) {
                tmp += " " + text.toLowerCase().charAt(i);
            } else if (text.charAt(i) != ' ') {
                tmp += text.charAt(i);
            }
        }

        return tmp;
    }

    public String combine() {
        String tmp = "";

        if (method == 'C')
            text = text.substring(0, 1).toUpperCase() + text.substring(1);
        else if (method == 'M')
            text += "()";

        tmp += text.charAt(0);

        for (int i = 1; i < text.length(); i++) {
            if (text.charAt(i - 1) == ' ') {
                tmp += text.toUpperCase().charAt(i);
            } else if (text.charAt(i) != ' ') {
                tmp += text.charAt(i);
            }
        }

        return tmp;
    }

    public String conversor() {
        String newText = "";

        if (operation == 'S')
            newText = strip();
        else if (operation == 'C')
            newText = combine();

        return newText;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        while (scanner.hasNextLine()) {
            String input = scanner.nextLine();
            int length = input.length();

            CamelCase4 solution = new CamelCase4();
            solution.setOperation(input.charAt(0));
            solution.setMethod(input.charAt(2));
            solution.setText(input.substring(4, length));

            String output = solution.conversor();
            System.out.println(output);
        }

        scanner.close();
    }
}